teps/0052-tekton-results-automated-run-resource-cleanup.md

```diff
@@ -73,26 +104,53 @@ Relevant Issues:
 
 ### Goals
 
+- Separate concerns for users who need current runtime data vs users who need
+  historical result data.
 - Define a mechanism for cleaning up completed Runs to minimize resources the
-  Pipeline controller is responsible for.
-- Preserve Run metadata so that users can continue to query it beyond the
-  lifetime of the Pipeline controller.
+  Pipeline data plane stores.
+- Preserve Run objects (e.g. the entire CRD) so that users can continue to query
+  it beyond the lifetime of the Pipeline data plane.
 
 ### Non-Goals
 
-- Providing a programmatic mechanism for end users to dynamically control deletion behavior.
+- Providing a programmatic mechanism for end users to dynamically control
+  deletion behavior.
+- Preservation of Pod logs.
+- Defining retention policies for Result API data.
 
 ### Use Cases (optional)
 
-This is a commonly requested feature due to the strain Run history this puts on
-etcd over time. See [Relevant Issues under Motivation](#motivation).
+When interacting with Run results, we see 2 primary users -
+
+1. Runtime Operators - Users that need to know what's currently running in a
+   cluster. They require the most up-to-date data, but don't require knowledge
+   of previous events when complete. Example users:
+
+   - Tekton Pipeline Controller
+   - Tekton Results Watcher
+   - Third Party Controllers responding to Task/PipelineRun lifecycle events.
+
+2. Result "Historians" (struggling with naming here) - Users who want to make
```

Comment:
we have some user profiles (https://github.com/tektoncd/community/blob/main/user-profiles.md) which might help, this sounds like it might be "Pipeline and Task Users" i.e. folks how are running stuff and want to see what happened

Reply:
Nice! I didn't know we had those- that will be useful in the future!

I think we still want custom users here, since many of those users would still fit into different roles depending on their use case.

- `Pipeline and Task Users` would generally fall under Historians, but there might be certain use cases (e.g. debugging a misbehaving pipeline) where it might be useful to dig into the currently running data plane.
- `Platform builders` / `Supporting tool developers` would need to be aware of both to know which API to integrate with based on their needs.

Reply:
i think if there are roles we are missing it'd be great to update https://github.com/tektoncd/community/blob/main/user-profiles.md with those roles vs. inventing new roles in each TEP - but i also dont think it's important enough to block this on

Reply:
I can look into adding these in another PR!